source/common/network/happy_eyeballs_connection_impl.cc

```diff
@@ -379,6 +380,27 @@ void HappyEyeballsConnectionImpl::dumpState(std::ostream& os, int indent_level)
   }
 }
 
+std::vector<Address::InstanceConstSharedPtr>
+HappyEyeballsConnectionImpl::sortAddresses(const std::vector<Address::InstanceConstSharedPtr>& in) {
+  std::vector<Address::InstanceConstSharedPtr> address_list = in;
+  for (auto current = address_list.begin(); current != address_list.end(); ++current) {
+    // Find the first address with a different address family than the current address.
+    auto it = std::find_if(current, address_list.end(), [&](const auto& val) {
+      return (val->type() != Address::Type::Ip || (*current)->type() != Address::Type::Ip ||
```

Comment:
kinda curious, does the spec say what to do with non IP addresses? Should we just remove those if we see any?